dandiapi/api/tests/test_asset.py

```diff
@@ -7,31 +8,97 @@
 import pytest
 import requests
 
-from dandiapi.api.models import Asset, AssetBlob
+from dandiapi.api.models import Asset, AssetBlob, Version
 
 from .fuzzy import HTTP_URL_RE, TIMESTAMP_RE, URN_RE, UUID_RE
 
+BASE_API_URL = 'https://api.dandiarchive.org'
+
 # Model tests
 
 
 @pytest.mark.django_db
 @pytest.mark.parametrize(
-    'path,qs,expected',
+    'path,asset_paths,expected',
     [
-        ('', [{'path': '/foo'}, {'path': '/bar/baz'}], ['bar/', 'foo']),
-        ('/', [{'path': '/foo'}, {'path': '/bar/baz'}], ['bar/', 'foo']),
-        ('////', [{'path': '/foo'}, {'path': '/bar/baz'}], ['bar/', 'foo']),
-        ('a', [{'path': '/a/b'}, {'path': '/a/c/d'}], ['b', 'c/']),
-        ('a/', [{'path': '/a/b'}, {'path': '/a/c/d'}], ['b', 'c/']),
-        ('/a', [{'path': '/a/b'}, {'path': '/a/c/d'}], ['b', 'c/']),
-        ('/a/', [{'path': '/a/b'}, {'path': '/a/c/d'}], ['b', 'c/']),
-        ('a', [{'path': 'a/b'}, {'path': 'a/c/d'}], ['b', 'c/']),
-        ('a', [{'path': 'a/b/'}, {'path': 'a/c/d/'}], ['b', 'c/']),
-        ('a', [{'path': '/a/b/'}, {'path': '/a/c/d/'}], ['b', 'c/']),
+        ('', ['foo', 'bar/baz'], {'folders': ['bar'], 'files': ['foo']}),
+        # ('', ['/foo', '/bar/baz'], {'folders': ['bar'], 'files': ['foo']}),  # edge case
```

Comment:
so this edge case shouldn't be tested/supported?

Reply:
This is commented out because while it ideally should be tested, it's difficult to achieve based on how this test is setup. I tried to fit this into what's currently there, but after a good deal of tinkering around, I was unsuccessful. The test itself does pass though, as I've done it manually, it's just hard to run alongside the other (more concrete and meaningful) tests.

I'm not really concerned about this, as this edge case shouldn't actually crop up in real use anyway, since we don't prefix asset paths with slashes. If we do care about this test case, I think we should track is separately from his PR, as I don't think it provides enough value to delay anything here.